stream.go

```diff
@@ -364,23 +369,32 @@ func (s *Stream) getNumBytesInReassemblyQueue() int {
 }
 
 func (s *Stream) onInboundStreamReset() {
-	var sid uint16
-	var resetOutbound bool
 	s.lock.Lock()
+	defer s.lock.Unlock()
+
 	s.log.Debugf("[%s] onInboundStreamReset: state=%s", s.name, s.state.String())
-	if s.state == StreamStateOpen {
-		s.state = StreamStateClosing
-		sid = s.streamIdentifier
-		resetOutbound = true
-	}
-	s.lock.Unlock()
 
-	// From draft-ietf-rtcweb-data-channel-13 section-6.7:
+	// No more inbound data to read. Unblock the read with io.EOF.
+	// This should cause DCEP layer (datachannel package) to call Close() which
+	// will reset outgoing stream also.
+
+	// See RFC 8831 section 6.7:
 	//	if one side decides to close the data channel, it resets the corresponding
 	//	outgoing stream.  When the peer sees that an incoming stream was
 	//	reset, it also resets its corresponding outgoing stream.  Once this
 	//	is completed, the data channel is closed.
-	if resetOutbound {
-		s.association.sendResetRequest(sid)
+
+	s.readErr = io.EOF
+	s.readNotifier.Broadcast()
+
+	if s.state == StreamStateClosing {
```

Comment:
If the state is not closing, do we change it to closing?

Reply:
No. At this point, peer closed it's outgoing stream, but this local node has not. (You can still send data to the peer.)

DCEP will see EOF, then close this stream by calling stream.Close(). That is when the state will go from "open" to "close".

I hope this state transition chart I shared with you earlier makes sense to you: [pion/sctp: State Transition](https://docs.google.com/spreadsheets/d/1sgQkvrL90vXiPEjJFECtgd95rukQL78z259bJLZE1RY/edit#gid=0)